src/test/java/PropositionalFormulaParserTest.java

```diff
@@ -0,0 +1,75 @@
+                new TestCase("DEFINED_HAS_ATTRIBUTE_&&HAS_ATTRIBUTE___LB__nonnull__RB__", new And(new Literal("DEFINED_HAS_ATTRIBUTE_"), new Literal("HAS_ATTRIBUTE___LB__nonnull__RB__"))),
+                new TestCase("HAS_BUILTIN___LB__nonnull__RB__&&A", new And(new Literal("HAS_BUILTIN___LB__nonnull__RB__"), new Literal("A"))),
+                new TestCase("A||(DEFINED___LB__NAME__RB__&&(NAME__GEQ__199630))", new Or(new Literal("A"), new And(new Literal("DEFINED___LB__NAME__RB__"), new Literal("NAME__GEQ__199630")))),
+                new TestCase("(DEFINED___LB__NAME__RB__&&(NAME__GEQ__199905)&&(NAME__LT__1991011))||(NAME__GEQ__300000)||DEFINED___LB__NAME__RB__", new Or(new And(new Literal("DEFINED___LB__NAME__RB__"), new And(new Literal("NAME__GEQ__199905"), new Literal("NAME__LT__1991011"))), new Or(new Literal("NAME__GEQ__300000"), new Literal("DEFINED___LB__NAME__RB__")))),
+                new TestCase("1__GT____U_MINUS__42", new Literal("1__GT____U_MINUS__42")),
+                new TestCase("1__GT____U_PLUS__42", new Literal("1__GT____U_PLUS__42")),
+                new TestCase("42__GT____U_TILDE__A", new Literal("42__GT____U_TILDE__A")),
+                new TestCase("A__ADD__B__GT__42", new Literal("A__ADD__B__GT__42")),
+                new TestCase("A__LSHIFT__B", new Literal("A__LSHIFT__B")),
+                new TestCase("A__THEN__B__COLON__C", new Literal("A__THEN__B__COLON__C")),
+                new TestCase("A__MUL____LB__B__ADD__C__RB__", new Literal("A__MUL____LB__B__ADD__C__RB__")),
+                new TestCase("(__LB____SQUOTE__Z__SQUOTE____SUB____SQUOTE__A__SQUOTE____RB____EQ__25)", new Literal("__LB____SQUOTE__Z__SQUOTE____SUB____SQUOTE__A__SQUOTE____RB____EQ__25")),
+                new TestCase("(__LB__GNUTLS_VERSION_MAJOR__ADD____LB__GNUTLS_VERSION_MINOR__GT__0__L_OR__GNUTLS_VERSION_PATCH__GEQ__20__RB____RB____GT__3)", new Literal("__LB__GNUTLS_VERSION_MAJOR__ADD____LB__GNUTLS_VERSION_MINOR__GT__0__L_OR__GNUTLS_VERSION_PATCH__GEQ__20__RB____RB____GT__3")),
+                new TestCase("(__LB__A__L_AND__B__RB____GT__C)", new Literal("__LB__A__L_AND__B__RB____GT__C")),
+                new TestCase("A__EQ__B", new Literal("A__EQ__B")),
+                new TestCase("(DEFINED_A)", new Literal("DEFINED_A")),
+                new TestCase("MACRO___LB__A__B__RB____EQ__1", new Literal("MACRO___LB__A__B__RB____EQ__1")),
+                new TestCase("ifndef", new Literal("ifndef")),
+                new TestCase("__HAS_WARNING___LB____QUOTE____SUB__Wa__SUB__warning__QUOTE_____foo__RB__", new Literal("__HAS_WARNING___LB____QUOTE____SUB__Wa__SUB__warning__QUOTE_____foo__RB__"))
+        );
+    }
+
+    /*
+     * A testCase is evaluated using syntactic equivalence between expectation and result because deterministic results are expected.
+     * Therefore, modifications in FeatureIDE might break these tests although the result remains semantically equivalent.
```

Comment:
They only remain semantically equivalent if the `CPPParserTest` succeeds, right? In theory, FeatureIDE could also introduce semantics-breaking changes (which would likely be a bug).